src/Bitpay/Client/Client.php

```diff
@@ -179,6 +183,179 @@ public function createInvoice(InvoiceInterface $invoice)
+            }
+
+            $data           = $body['data'];
+            $supportRequest = new SupportRequest();
+            $supportRequest->setId($data['id'])
+                ->setRequestDate($data['requestDate'])
+                ->setStatus($data['status'])
+                ->setToken($data['token']);
+
+            return $supportRequest;
+        } catch (\Exception $ex) {
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getRefund($invoiceId, $refundRequestId)
+    {
+        try {
+            $invoice = $this->getInvoice($invoiceId);
+        } catch (\Exception $ex) {
+            throw $ex;
+        }
+
+        $request = $this->createNewRequest();
+        $request->setMethod(Request::METHOD_GET);
+        $request->setPath(sprintf('invoices/%s/refunds/%s?token=%s', $invoiceId, $refundRequestId, $invoice->getToken()->getToken()));
+
+        try {
+            $this->addIdentityHeader($request);
+            $this->addSignatureHeader($request);
+
+            $this->request  = $request;
+            $this->response = $this->sendRequest($request);
+            $body           = json_decode($this->response->getBody(), true);
+
+            if (isset($body['error'])) {
+                throw new \Exception($body['error']);
+            }
+
+            $data           = $body['data'];
+            $supportRequest = new SupportRequest();
+            $supportRequest->setId($data['id'])
+                ->setRequestDate($data['requestDate'])
+                ->setStatus($data['status'])
+                ->setToken($data['token']);
+
+            return $supportRequest;
+        } catch (\Exception $ex) {
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function cancelRefund($invoiceId, $refundRequestId)
+    {
+        try {
+            $refund = $this->getRefund($invoiceId, $refundRequestId);
+        } catch (\Exception $ex) {
+            throw $ex;
+        }
+
+        $request = $this->createNewRequest();
+        $request->setMethod(Request::METHOD_DELETE);
+        $request->setPath(sprintf('invoices/%s/refunds/%s', $invoiceId, $refundRequestId));
+
+        try {
+            $body = [
+                'token' => $refund->getToken()
+            ];
+
+            $request->setBody(json_encode($body));
+
+            $this->addIdentityHeader($request);
+            $this->addSignatureHeader($request);
+
+            $this->request  = $request;
+            $this->response = $this->sendRequest($request);
+            $body           = json_decode($this->response->getBody(), true);
+
+            if (isset($body['error'])) {
+                throw new \Exception($body['error']);
+            }
+
+            return $body['data'];
+        } catch (\Exception $ex) {
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getRefunds($invoiceId)
```

Comment:
There is a lot of duplicate logic in these 4 methods, they should be broken down and simplified.